cpp/open3d/visualization/gui/NativeLinux.cpp

```diff
@@ -18,20 +19,31 @@ namespace visualization {
 namespace gui {
 
 void* GetNativeDrawable(GLFWwindow* glfw_window) {
-    return (void*)glfwGetX11Window(glfw_window);
+    const int platform = glfwGetPlatform();
+    if (platform == GLFW_PLATFORM_X11) {
+        return (void*)glfwGetX11Window(glfw_window);
+    } else if (platform == GLFW_PLATFORM_WAYLAND) {
+        return (void*)glfwGetWaylandWindow(glfw_window);
+    } else {
+        return nullptr;
+    }
 }
 
 void PostNativeExposeEvent(GLFWwindow* glfw_window) {
-    Display* d = glfwGetX11Display();
-    auto x11win = glfwGetX11Window(glfw_window);
+    const int platform = glfwGetPlatform();
 
-    XEvent e;
-    memset(&e, 0, sizeof(e));
-    e.type = Expose;
-    e.xexpose.window = x11win;
+    if (platform == GLFW_PLATFORM_X11) {
+        Display* d = glfwGetX11Display();
+        auto x11win = glfwGetX11Window(glfw_window);
 
-    XSendEvent(d, x11win, False, ExposureMask, &e);
-    XFlush(d);
+        XEvent e;
+        memset(&e, 0, sizeof(e));
+        e.type = Expose;
+        e.xexpose.window = x11win;
+
+        XSendEvent(d, x11win, False, ExposureMask, &e);
+        XFlush(d);
+    }
```

Comment:
Is there a corresponding action needed in Wayland? I understand this is telling X11 to redraw the window.

Reply:
I don't know why this is done manually here and not via GLFW. I would naively assume that this is not required here and that the X11 support would work without it.

In any case, there is no such thing as an `Expose` event on Wayland.

Reply:
I just removed the code in `PostNativeExposeEvent` and ran `ViewGeometry` on `Xwayland` without issues. So I guess this is redundant.

Reply:
I'm not sure that this code is called by `DrawGeometries`. We have two visualizers in Open3D - the legacy / pure OpenGL one (`DrawGeometries`) and the new filament + GUI visualizer (the code in the gui and rendering folders) - this is called by `draw()` and `O3DVisualizer`.

Reply:
I didn't know that there are two different visualiser implementations. That explains why the GUI examples behave differently.